mersenne-31/src/extension.rs

```diff
@@ -0,0 +1,80 @@
+use p3_field::extension::cubic::CubicOef;
+use p3_field::extension::quadratic::QuadraticOef;
+use p3_field::extension::OptimallyExtendable;
+
+use crate::{Mersenne31, Mersenne31Complex};
+
+impl OptimallyExtendable<2> for Mersenne31Complex<Mersenne31> {
+    type Extension = QuadraticOef<Self>;
+
+    // Verifiable in Sage with
+    // `p = 2**31 - 1  # Mersenne31`
+    // `F = GF(p)  # The base field GF(p)`
+    // `R = F['x']  # The polynomial ring over F`
+    // `K = F.extension(R.gen()**2 + 1, 'i')  # The extension field of complex`
+    // `R2 = K['y']`
+    // `f2 = R2.gen()**2-K('i')-2   # polynomial y^2-i-2`
+    // `f.is_irreducible()`
```

Comment:
```suggestion
    // Verifiable in Sage with
    // ```sage
    // p = 2**31 - 1  # Mersenne31
    // F = GF(p)  # The base field GF(p)
    // R.<x> = F[]  # The polynomial ring over F
    // K.<i> = F.extension(x^2 + 1)  # The complex extension field
    // R2.<y> = K[]
    // f2 = y^2 - i - 2 
    // assert f2.is_irreducible()
    // ```
```